wled00/FX.cpp

```diff
@@ -10644,6 +10644,118 @@ uint8_t WS2812FX::addEffect(uint8_t id, mode_ptr mode_fn, const char *mode_name)
   }
 }
 
+/*
+  Simple bit strobe
+*/
+uint16_t mode_music_strobe(void) {
+  um_data_t *um_data = getAudioData();
+  if (!um_data) return FRAMETIME; 
+
+  bool samplePeak = *(bool*)um_data->u_data[3]; 
+  int volumeRaw   = *(int16_t*)um_data->u_data[1]; 
+
+  long mappedSpeed = map(SEGMENT.speed, 0, 255, 0, 20);
+  uint8_t fadeBy = 255 - mappedSpeed;
+  SEGMENT.fade_out(fadeBy);
+
+  uint8_t threshold = SEGMENT.intensity;
+
+  if (samplePeak && volumeRaw > threshold) {
+    SEGMENT.fill(SEGCOLOR(0));
+  }
+  
+  return FRAMETIME;
+}
+
+static const char _data_FX_MODE_MUSIC_STROBE[] PROGMEM = "Music Strobe@Fade speed,Threshold;!,!;!;1v;ix=128,sx=224,si=1";
+
+/*
+  Flame jet
+*/
+uint16_t mode_flame_jet(void) {
+  um_data_t *um_data = getAudioData();
+  if (!um_data) return FRAMETIME;
+
+  bool samplePeak = *(bool*)um_data->u_data[3];
+  int volumeRaw   = *(int16_t*)um_data->u_data[1];
+
+  long mappedFade = map(SEGMENT.intensity, 0, 255, 0, 64);
+  uint8_t fade_by = 255 - mappedFade;
+  SEGMENT.fade_out(fade_by);
+
+  uint8_t threshold = SEGMENT.custom1;
+
+  if (samplePeak && volumeRaw > threshold && SEGENV.aux0 == 0) {
+    SEGENV.aux0 = 1; // start animation
+  }
+
+  // --- Animation ---
+  if (SEGENV.aux0 > 0) {
+    long mappedSpeed = map(SEGMENT.speed, 0, 255, 0, 32);
+    uint16_t rise_amount = 1 + (uint16_t)(volumeRaw * (1 + (uint16_t)mappedSpeed)) / 2000;
+
+    if (SEGMENT.is2D()) {
+      // --- 2D LOGIC ---
+      const uint16_t max_height = SEG_H;
+      uint16_t start_y = SEG_H - 1;
+      
+      for (uint16_t y_offset = 0; y_offset < SEGENV.aux0; y_offset++) {
+        uint16_t current_y = start_y - y_offset;
+        if (current_y >= SEG_H) continue; // Boundary check
+
+        uint32_t base_color = SEGCOLOR(0);
+        CRGB fastled_color = CRGB(base_color);
+
+        uint8_t heat_reduction = map(y_offset, 0, SEG_H, 0, 180);
+        fastled_color.g = qsub8(fastled_color.g, heat_reduction);
+        fastled_color.b = qsub8(fastled_color.b, heat_reduction * 2);
```

Comment:
What happens when one of the color channels is less than ``heat_reduction`` ? Will the value wrap around (= higher value), or will it be clamped at 0?